File: backend/components/metiers/administration_publique/utils/plugins/core/pluginManager.test.py
"""
pluginManager.test.py – Tests unitaires Python pour pluginManager Threed
"""
# Fichier déplacé dans samples/pluginManager.test.py
from .pluginManager import PluginManager

def test_plugin_manager_run_all():
    pm = PluginManager()
    pm.register(lambda data: {'status': 'processed'})
    results = pm.run_all({'input': 'test'})
    assert any(r.get('status') == 'processed' for r in results)

File: backend/components/metiers/administration_publique/utils/plugins/core/pluginManager.test.js
// Fichier déplacé dans samples/pluginManager.test.js
// pluginManager.test.js – Tests unitaires JS pour pluginManager Threed
const { pluginManager } = require('./pluginManager');
describe('pluginManager Threed', () => {
  it('doit exécuter tous les plugins', () => {
    const result = pluginManager.runAll({ input: 'test' });
    expect(result.length).toBeGreaterThan(0);
  });
});
